packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -140,6 +140,15 @@ class InteractiveViewer extends StatefulWidget {
   /// If set to false, then the child will be given infinite constraints. This
   /// is often useful when a child should be bigger than the InteractiveViewer.
   ///
+  /// For example, for a child which is bigger than the viewport but can be
+  /// panned to reveal parts that were initially offscreen, constrained must be
+  /// set to false to allow it to size itself properly. If constrained is true
+  /// and the child can only size itself to the viewport, then areas initially
+  /// outside of the viewport will not be able to receive user interaction
```

Comment:
This behavior sounds a little bit unintuitive. Is this a current limitation or this is by design?

Reply:
This is a quirk of Flutter because:

  1. Flutter by default allows widgets to draw outside of their `size` without being clipped.
  1. Flutter doesn't send events to the widget if they happen on these "overflowing" parts.

In both cases I've seen this problem, it was when using a CustomPaint in InteractiveViewer and drawing something bigger than the screen.  The widget was "constrained" to the size of the screen even though it was drawing beyond it.  By setting the `constrained` parameter to false, InteractiveViewer gives its child infinite constraints, so the child can be as big as it wants.

Reply:
Ah this line I see: https://github.com/flutter/flutter/blob/c706abf07526aa40a456b32da51fed6b111a2bf8/packages/flutter/lib/src/rendering/box.dart#L2234

Reply:
Exactly, I tracked it to that same line awhile back.